src/core/highlight-vars.js

```diff
@@ -0,0 +1,22 @@
+// Module core/highlight-vars
+// Highlights occurrences of a <var> in an algorithm on click
+// if a <var> is set only once (unused/undeclared), it's highlighted as "var.bug"
+
+export const name = "core/highlight-vars";
+
+export function run(conf, doc, cb) {
+  const vars = [...document.querySelectorAll(".algorithm var")];
+  for (let i = 0, l = vars.length; i < l; ++i) {
+    vars[i].addEventListener("click", highlightVars);
+  }
+  cb();
+}
+
+function highlightVars({ target }) {
+  const value = target.innerText;
+  const parent = target.closest(".algorithm");
+  const toHightlight = [...parent.querySelectorAll("var")]
+    .filter(el => el.innerText === value);
+  const classToAdd = toHightlight.length === 1 ? "bug" : "active";
+  toHightlight.map(el => el.classList.toggle(classToAdd));
```

Comment:
This isn't really a `map`... maybe just:

```JS
[...parent.querySelectorAll("var")]
  .filter(el => el.innerText === value);
  .forEach((el, _, {length}) => {
     const classToAdd = length === 1 ? "bug" : "active";
     el.classList.toggle(classToAdd);
   });
```

However, we have a problem when exporting (as we can't export JS). We either need to see if there is a way to do this with just CSS, or we need to make sure the CSS for this is not exported out.

Reply:
1. About doing in CSS: we can use `[data-var="value"]` as selector. That will need adding a style for each variable value (during export) on page. That will make page stylesheet really long. Should I do it?
2. We can remove the `var` CSS in export altogether by adding `.removeOnSave` on stylesheet for `var`, which is really simple thing to do but removes the highlight functionality in exports.

I'll try to find out more ways for CSS only part.

Reply:
I tried the first approach, and it won't work, as we can't highlight (tried on `:hover`) all the vars with same `data-var` as CSS has no parent selector.
Why don't we export JS though?

Reply:
The W3C doesn’t allow us to. The only JS allowed in specs is “fixup.js”. However, if we prove this would be really useful, we can add the JS to fixup.js.

Reply:
For now, I'll just remove the highlight functionality from exported content, if that's fine.

Reply:
Let’s not bloat the css. Let’s just start with this JS version to show it’s useful.